src/components/CollectionsAndPalettes/common/RoundButton/RoundButton.tsx

```diff
@@ -1,45 +1,155 @@
 import * as React from "react";
 import cn from "classnames";
-import { AppContext, AppContextState } from "../../../../context/AppContext";
 import { Link } from "react-router-dom";
-import { COLLECTIONS_URL, PALETTES_URL } from "../../../../constants/urls";
+import {
+  COLLECTIONS_URL,
+  PALETTES_URL,
+  COLLECTIONS_EDIT_URL,
+  COLLECTIONS_INFO_URL,
+  COLLECTIONS_IMAGE_URL,
+} from "../../../../constants/urls";
 
 interface Props {
   buttons: any;
+  explore?: boolean;
+  activeMode?: string;
+  activeDisplay?: string;
   background?: string;
+  iconClass?: any;
+  hasAnimation?: boolean;
+  toggleMode?: any;
+  toggleDisplay?: any;
 }
 
-const activeButtons = {
+const activeButtonURLS = {
   palettes: PALETTES_URL,
   collections: COLLECTIONS_URL,
+  image: COLLECTIONS_IMAGE_URL,
+  info: COLLECTIONS_INFO_URL,
+  edit: COLLECTIONS_EDIT_URL,
 };
 
 export default class RoundButton extends React.Component<Props, any> {
-  static contextType = AppContext;
-  context!: AppContextState;
+  activeButtonMode: number;
+  animate = null;
+  animationDirection = null;
+
+  constructor(props) {
+    super(props);
+    switch (this.props.activeMode) {
+      case "image": {
+        this.activeButtonMode = 0;
+        break;
+      }
+      case "info": {
+        this.activeButtonMode = 1;
+        break;
+      }
+      case "edit": {
+        this.activeButtonMode = 2;
+        break;
+      }
+    }
+  }
+
+  switchDisplay(e, index, button) {
+    this.props.toggleDisplay(button);
+  }
+
+  toggleAnimation(e, index, button) {
+    this.animate = "true";
```

Comment:
I think there should be a way easier way of getting this effect. Instead of adding global animations for this, you can just add a class to the button that adds 
```
position: absolute;
left: 0;
transition: left 0.25s ease-in-out;
maybe some z-index stuff if needed
```
and then just control the `left` property from the component, based on the index of the selected button:
```
<button
  ...restOfProps
  style={{
    left: `${(currentIndex / nrOfButtons) * 100}%`
  }}
>
```